src/encoded/static/components/motifs.js

```diff
@@ -0,0 +1,144 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+// import * as logos from '../libs/d3-sequence-logo'; // This is for local development when changes are needed to d3-sequence-logo.
+
+// Convert PWM file into JavaScript object
+// Input "str" consists of PWM file
+// Each row corresponds to a position in the motif
+// The farthest left column is an index (no important nucleotide information)
+// The next three columns describe the frequency of occurrence of each nucleotide
+// Columns correspond from left to right to nucleotides A, C, G, and T
+function convertTextToObj(str) {
+    // Split the file by row and create a new object
+    // The new object has an element corresponding to each row and each element is an array of the entries in that row, split by tab (the PWM file is tab-delineated)
+    const cells = str.split('\n').map(el => el.split(/\s+/));
+    const obj = [];
+    // Create a new object without the index or rows that contain comments or unnecessary information and convert the frequency values from strings to numbers
+    cells.forEach((cell) => {
+        if (cell.length === 6) {
+            obj.push([+cell[1], +cell[2], +cell[3], +cell[4]]);
+        }
+    });
+    // We will use "obj" to create the logos because it has all the important nucleotide frequency information from the PWM file
+    return obj;
+}
+
+// Fetch PWM file
+export function getMotifData(pwmLink, fetch) {
+    return fetch(pwmLink, {
+        method: 'GET',
+        headers: {
+            Accept: 'application/json',
+        },
+    }).then((response) => {
+        if (response.ok) {
+            return response.text();
+        }
+        throw new Error('not ok');
+    }).catch((e) => {
+        console.log('OBJECT LOAD ERROR: %s', e);
+    });
+}
+
+// Display information on page as JSON formatted data
+export class Motifs extends React.Component {
+    constructor() {
+        super();
+
+        this.drawMotifs = this.drawMotifs.bind(this);
+        this.addMotifElement = this.addMotifElement.bind(this);
+    }
+
+    componentDidMount() {
+        require.ensure(['d3', 'd3-sequence-logo'], (require) => {
+            if (this.chartdisplay) {
+                this.d3 = require('d3');
+                this.sequenceLogos = require('d3-sequence-logo'); // logos This is for local development when changes are needed to d3-sequence-logo.
+                const targetElement = this.chartdisplay;
+                this.drawMotifs(targetElement);
+            }
+        });
+    }
+
+    // Redraw charts when window changes
+    componentDidUpdate() {
+        if (this.chartdisplay) {
+            this.chartdisplay.innerHTML = '';
+            const targetElement = this.chartdisplay;
+            this.drawMotifs(targetElement);
+        }
+    }
+
+    addMotifElement(targetElement, response, d, entryPoint) {
```

Comment:
Without spending the time that you have on this, I could be completely off, but I feel like a lot of direct DOM manipulation could be avoided in this method by using a state set by getMotifData() to render these divs in `render()`, and some of these divs use props, so they wouldn’t even need to use state. The only exception would be entryPoint which couldn’t be called until after mount, but you can get to the divs that it renders into through refs, or through getElementById in the rare case that refs can't handle this.

Reply:
I'm not sure if I did exactly the strategy you suggested, but I moved as much code as possible into render() and generally wrote things in a more React way.